vspace/src/main/java/edu/asu/diging/vspace/core/factory/impl/ImageFactory.java

```diff
@@ -23,4 +35,28 @@ public IVSImage createImage(String filename, String fileType) {
         image.setFileType(fileType);
         return image;
     }
+    
+    /**
+     * Method to rename image   
+     * 
+     * @param imageId - image unique identifier
+     * @param imageForm - ImageForm with updated values for image fields
+     * @return throws FileNotFoundException if no image exists with id, 
+     * throws FileStorageException if file renaming fails 
+     */ 
+    @Override
+    public void editImage(String imageId, ImageForm imageForm) throws FileNotFoundException, FileStorageException {
+        if (imageRepo.findById(imageId).isPresent()) {
+            VSImage image = imageRepo.findById(imageId).get();
+            if (storage.renameImage(image, imageForm.getFileName())) {
```

Comment:
filenames do not need to be changable at this moment, only name and description.